### PR TITLE
[Master] Exit RunBuildToolsTask upon tool error

### LINKED ISSUE #42483: [Improvement]: Avoid the package resolution and the compilation when there is an error diagnostic from the tool
### Description

The bal tools continue to do package resolution and compilation even after some error diagnostics returned from the tool execution.

### Describe your problem(s)

With the tool integration with `bal build/test/run` commands, the tool is getting engaged and some codes are generated. Due to specific reason the tools can opt-in to add a error diagnostic via the tool-context notifying that the tool execution has failures.

The current implementation print those diagnostics but anyway continue to package resolution and compilation which prints some compiler errors which does not represents the actual issue.

### Describe your solution(s)

This behaviour can be improved by checking the diagnostics before executing package resolution and compilation and if there are any errors terminate the process.

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by Thevakumar-Luheerathan**
Fixes #42483 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunBuildToolsTask.java [hunk 1]
```diff
@@ -207,6 +211,17 @@ public void execute(Project project) {
...
+
+        // Exit if there is any error diagnostic
+        boolean hasErrors = false;
+        for (Diagnostic d : toolDiagnostics) {
+            if (d.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR)) {
+                hasErrors = true;
+            }
+        }
+        if (hasErrors) {
+            throw createLauncherException("compilation contains errors");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunBuildToolsTask.java:223]:
```suggestion
            throw createLauncherException("build tool execution contains errors");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunBuildToolsTask.java [hunk 2]
```diff
@@ -207,6 +211,17 @@ public void execute(Project project) {
...
             }
         }
+
+        // Exit if there is any error diagnostic
+        boolean hasErrors = false;
+        for (Diagnostic d : toolDiagnostics) {
+            if (d.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR)) {
+                hasErrors = true;
+            }
+        }
```
Comment (TharmiganK) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunBuildToolsTask.java:221]: We can simply this using functional style expression:
```suggestion
boolean hasErrors = toolDiagnostics.stream()
    .anyMatch(d -> d.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR));
```

File: cli/ballerina-cli/src/test/resources/test-resources/command-outputs/unix/build-bal-project-with-build-tool-not-found.txt [hunk 3]
```diff
@@ -1,6 +1,3 @@
 
 Executing Build Tools
 ERROR [Ballerina.toml:(6:1,9:26)] Build tool 'grpc' cannot be resolved
-
```
Comment (azinneera) [inline cli/ballerina-cli/src/test/resources/test-resources/command-outputs/unix/build-bal-project-with-build-tool-not-found.txt:4]: Isn't the `build tool execution contains errors` part getting printed?

**Review by gayaldassanayake — APPROVED** [review]

File: cli/ballerina-cli/src/test/resources/test-resources/command-outputs/unix/build-bal-project-with-build-tool-not-found.txt (hunk 3 above)
Comment (Thevakumar-Luheerathan) [reply cli/ballerina-cli/src/test/resources/test-resources/command-outputs/unix/build-bal-project-with-build-tool-not-found.txt:4]: It is printed properly.

**Pull request merged**